### PR TITLE
Revert GameEntity and fix tests

### PR DESCRIPTION
**Pull request by wilcorg**
make ORM great again!

### CODE REVIEW

**Review by imnotpoz — COMMENTED** [review]
why did you remove the name field in @PrimaryKeyJoinColumn?

File: src/main/java/org/gogame/server/domain/entities/LeaderboardEntity.java [hunk 1]
```diff
@@ -14,6 +14,7 @@
 public class LeaderboardEntity {
 
     @Id
+    @GeneratedValue(strategy = GenerationType.SEQUENCE)
     @Column(name = "user_pos", nullable = false, unique = true)
     private Long userPos;
```
Comment (imnotpoz) [inline src/main/java/org/gogame/server/domain/entities/LeaderboardEntity.java]: isn't this the user's position on the leaderboard? why is it autogenerated?

File: src/main/java/org/gogame/server/domain/entities/dto/GameMessageDto.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+public class GameMessageDto {
+
+    private Long author;
+
+    private Long userSenderId;
+
+    private Long userReceiverId;
```
Comment (imnotpoz) [inline src/main/java/org/gogame/server/domain/entities/dto/GameMessageDto.java]: get rid of both `userSenderId` and `userReceiverId`, add a `gameId` and change `author` to `authorId`

**Comment by wilcorg** [comment]
> why did you remove the name field in @PrimaryKeyJoinColumn?

`name` arg in `@PrimaryKeyJoinColumn` doesn't work
20240126_17h36m04s_grim
Relations are overriding suggested names
PS: everything is actual and was refreshed before screenshot

File: src/main/java/org/gogame/server/domain/entities/LeaderboardEntity.java (hunk 1 above)
Comment (wilcorg) [reply src/main/java/org/gogame/server/domain/entities/LeaderboardEntity.java]: `userPos` is completely screwed up and should be removed. Thanks for issuing that

File: src/main/java/org/gogame/server/domain/entities/dto/GameMessageDto.java (hunk 2 above)
Comment (wilcorg) [reply src/main/java/org/gogame/server/domain/entities/dto/GameMessageDto.java]: I rewrote it from your dto specifications. I will change it in the next PR

**Pull request merged**